sklearn/base.py

```diff
@@ -332,10 +336,19 @@ def _get_tags(self):
         return collected_tags
 
 
+def _sizeof(cls):
+    #  return sum(x.nbytes if hasattr(x, 'nbytes') else sys.getsizeof(x)
+    #  for x in cls.__dict__.values())
```

Comment:
Does the previous version not work?
Or is `sys.getsizeof` using x.nbytes when available? I would have imagined a specialized handling maybe for ndarray, csr/csc arrays and DataFrames ?

Reply:
I've uncommented this version (and added another check for `not callable(x.nbytes)`). `x.nbytes` is a better approximation than `sys.getsizeof`:

``` python
>>> N = 1_000_000
>>> x = np.arange(N).reshape(-1, N)
>>> x.shape
(1, 1000000)
>>> sys.getsizeof(x)
112
>>> x.nbytes
8000000
>>> sys.getsizeof(x.flatten())
8000096
```